perun-core/src/main/java/cz/metacentrum/perun/core/bl/MembersManagerBl.java

```diff
@@ -291,11 +290,10 @@
 	 *
 	 * @return non-sponsored member
 	 *
-	 * @throws MemberNotExistsException If member not exists by defined id in system Perun
 	 * @throws MemberNotSponsoredException If member was not set as sponsored before calling this method.
 	 * @throws InternalErrorException if something unexpected happend
 	 */
-	Member unsetSponsorshipForMember(PerunSession session, Member sponsoredMember) throws MemberNotExistsException, MemberNotSponsoredException, InternalErrorException;
```

Comment:
This kind of removal of not-thrown Exceptions (while I like it in principle) worries me. Since if implementation fails in this case, it throws InternalErrorException with SQL Exception inside. Its OK, but I'm not sure if explicit check on member existence was not intended and is actually missing in implementation! Probably depends on what we want to see in logs/GUI - MemberNotExistsException or InternalErrorException.